department-of-veterans-affairs/caseflow
# WIP: Shalliburton/caseflow 417 hearing day time slots

Resolves [CASEFLOW-417](https://vajira.max.gov/browse/CASEFLOW-417)

### Description
Adds time slot details to hearing days and displays those details in the UI for confirmation from the user building the hearing schedule

### Acceptance Criteria
- [ ] Build Schedule spreadsheet updated to build Video/In-Person days or Virtual hearing days as described in https://docs.google.com/document/d/1O94ifyf-uhp-g9iCGz1zIeCxis-E4SBfFJDZnp5gO-0/edit
- [ ] Add columns for time slot rules in the confirmation step of the Build Schedule workflow
- [ ] Write values entered in the new spreadsheet columns into the database columns created in (CASEFLOW-1115)
- [ ] All columns require a value
- [ ] Number of time slots can't exceed 12
- [ ] Duration of time slots can't exceed 60
- [ ] Start time has to be able to be interpreted as a time (but flexible on other input requirements)
- [ ] 🧘‍♂️ Can we add dropdowns with limited values into the template itself?

### Testing Plan

1. Sign in as user `BVASYELLOW` http://localhost:3000/test/users
2. Navigate to the build hearings schedule: http://localhost:3000/hearings/schedule/build
3. Click the "Upload Files" button
4. Fill out the start date as 04/01/2021 and the end date as 06/30/2021
5. Upload the spreadsheet: [RO Assignment Test Q3 FY21 .xlsx](https://github.com/department-of-veterans-affairs/caseflow/files/6423512/RO.Assignment.Test.Q3.FY21.xlsx)
6. Click continue to schedule the assignments
7. Confirm the time slot details and then click confirm assignments and confirm when the dialog pops up
8. Ensure that you get a success message indicating the hearing days were created

### User Facing Changes

|Before|After|
|--|--|
|<img width="1719" alt="Screen Shot 2021-05-04 at 12 40 49 PM" src="https://user-images.githubusercontent.com/61989526/117060654-fac29600-acd5-11eb-958d-d39f0051eede.png">|<img width="1704" alt="Screen Shot 2021-05-04 at 12 32 41 PM" src="https://user-images.githubusercontent.com/61989526/117060489-d6ff5000-acd5-11eb-8769-681cb65a268a.png">|
